homeassistant/components/planifneige/sensor.py

```diff
@@ -0,0 +1,132 @@
+"""
+Sensor for the City of Montreal's Planif-Neige snow removal APIs.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.planifneige/
+"""
+
+import logging
+
+from homeassistant.const import ATTR_ATTRIBUTION
+from homeassistant.helpers.restore_state import RestoreEntity
+
+DEPENDENCIES = ['planifneige']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_STREETID = 'streetid'
+CONF_STREETS = 'streets'
+CONF_DBPATH = 'database_path'
+
+ATTR_SENSOR = 'sensor'
+ATTR_STATES = 'states'
+
+DOMAIN = 'planifneige'
+
+STREET_STATE = {
+    '0': ['Snowed', 'mdi:snowflake'],  # street is snowed in
+    '1': ['Clear', 'mdi:road'],  # street is clear
+    '2': ['Planned', 'mdi:clock-outline'],  # street is planned for clearing
+    '3': ['Planned', 'mdi:clock-outline'],  # clearing planned with new date
+    '4': ['Snowed', 'mdi:snowflake'],  # to be planned, still snowed in
+    '5': ['Clearing', 'mdi:bulldozer'],  # trucks loading, still clearing
+    '10': ['Ploughed', 'mdi:road']  # street cleared, snow not loaded
+}
+
+PLANIFNEIGE_ATTRIBUTION = "Information provided by the City of Montreal "\
+    "(http://donnees.ville.montreal.qc.ca/dataset/deneigement)"
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the PlanifNeige platform."""
+    data = hass.data[DOMAIN]
+    async_add_entities(
+        [PlanifNeigeSensor(data, sensor) for sensor in discovery_info]
+    )
+
+
+class PlanifNeigeSensor(RestoreEntity):
+    """PlanifNeige sensor."""
+
+    def __init__(self, data, sensor):
+        """Initialize the sensor."""
+        self._data = data.data
+        self._state = None
+        self._name = sensor['name']
+        self._streetid = sensor['streetid']
+        self._icon = ""
+        self._start_plan_date = None
+        self.__end_plan_date = None
+        self._start_replan_date = None
+        self._end_replan_date = None
+        self._date_updated = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def streetid(self):
+        """Return the name of the sensor."""
```

Comment:
Please adapt this docstring (and others below) to refer to its actual property. Don't worry, I tend to overlook these also. 😉